tests/spec/fs.mkdir.spec.js

```diff
@@ -46,3 +46,51 @@ describe('fs.mkdir', function() {
     });
   });
 });
+
+describe('fs.promises.mkdir', function () {
+  beforeEach(util.setup);
+  afterEach(util.cleanup);
+
+  it('should be a function', function () {
+    var fs = util.fs();
+    expect(fs.promises.mkdir).to.be.a('function');
+  });
+
+  it('should return an error if part of the parent path does not exist', function (done) {
+    var fs = util.fs();
+
+    fs.promises.mkdir('/tmp/mydir', function (error) {
```

Comment:
This is mixing callbacks and promises.  You'd need to change to something like this:

```js
fs.promises.mkdir('/tmp/mydir', () =>
  return fs.promises
    .mkdir('/tmp/mydir')
    .catch(error => {
      expect(error).to.exist);
      expect(error.code).to.equal('ENOENT');
    });
);
```